lint/problem.go

```diff
@@ -46,3 +48,65 @@ type Problem struct {
 
 	noPositional struct{}
 }
+
+// Problems is a slice of individual Problem objects.
+type Problems []Problem
+
+// Equal determines whether a Problem is sufficiently similar to another
+// to be considered equal.
+//
+// This is intended for unit tests and is intentially generous on what
+// constitutes equality.
+func (problems Problems) Equal(other Problems) bool {
+	// `other` may be nil.
+	// Consider an length-0 slice to be equal to nil.
```

Comment:
Go already considers this to be true, you can remove this condition.

len(nil) == 0

Reply:
It very much does not. :-) I had numerous problems returning `nil` and having `reflect.DeepEqual` complain about lack of equivalence to `[]Problem`.

Reply:
https://play.golang.org/p/NjLpZJVZRTA ?

Reply:
`reflect.DeepEqual` does check for `nil`-ness:
> Slice values are deeply equal when all of the following are true: they are both nil or both non-nil, they have the same length, and either they point to the same initial entry of the same underlying array (that is, &x[0] == &y[0]) or their corresponding elements (up to length) are deeply equal. Note that a non-nil empty slice and a nil slice (for example, []byte{} and []byte(nil)) are not deeply equal.

But that's not what you want in this case, you're assuming if something is nil that it's equivalent to having `len(.) == 0`, which is already true.

Reply:
Confirmed tests still pass with what I have now. Special `nil` behavior excised.